modules/andes-core/broker/src/main/java/org/wso2/andes/mqtt/MQTTopicManager.java

```diff
@@ -208,7 +208,7 @@ public void removeOrDisconnectTopicSubscription(String mqttClientChannelID, Subs
                 //Will remove the subscriber cluster wide
                 try {
                     //Will indicate the disconnection of the topic
-                    if (action == SubscriptionEvent.DISCONNECT) {
+                    if (action == SubscriptionEvent.DISCONNECT && !(subscriber.getQOSLevel() == 0 && !subscriber.isCleanSession())) {
```

Comment:
There's a enum defining qos levels. can you use that enum with .equals.
ex:- subscription.getQOSLevel().equals(QOSLevelEnum.AT_MOST_ONCE)